### PR TITLE
#167190455 bookmark article for reading later

### PR DESCRIPTION
**Pull request by obayomi96**
#### What does this PR do?
- Implement bookmarking of an article for reading later

#### Description of Task to be completed?
- Create bookmark Model and properly associate it with relevant models
- Create Migrations.
- Write test
- Implement Bookmark single article
- Implement unbookmark single article
- Implement get all bookmarked articles
- Implement unbookmark all bookmarked articles
#### How should this be manually tested?
- Clone the repo
- Run migrations with the following command
`npm run migrate`
- Start the server
- Go to the endpoint below
- To get all bookmarked articles
`GET api/v1/user/bookmark` 
- To bookmark an article
`POST api/v1/user/bookmark/:articleId`
- To unbookmark an article
`DELETE api/v1/user/bookmark/:articleId`
- To unbookmark all bookmarked articles
`DELETE api/v1/user/bookmark`

#### Any background context you want to provide?
- None

#### What are the relevant pivotal tracker stories?
#167190455

#### Screenshots (if appropriate)
Screenshot 2019-08-07 at 7 41 58 PM
Screenshot 2019-08-07 at 7 42 08 PM
Screenshot 2019-08-07 at 7 42 17 PM

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
   "main": "index.js",
   "scripts": {
     "start": "node build/index.js",
-    "dev": "export NODE_ENV=production && DEBUG=dev && nodemon --exec babel-node server/index.js",
+    "dev": "export DEBUG=dev && nodemon --exec babel-node server/index.js",
```
Comment (chuxmykel) [inline package.json:8]: 👍🏾

File: server/controllers/bookmarkController.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+   * @returns {Object} Success Message
+   */
+  static async unbookmarkArticle(req, res) {
+    const userId = req.user.id;
+    const { articleId } = req.params;
+    const article = await models.Article.findByPk(articleId);
+    if (!article) return utils.errorStat(res, 404, 'Article not found');
+    const bookmarked = await models.Bookmark.findOne({
+      where: {
+        [Op.or]: [{ articleId }]
```
Comment (chuxmykel) [inline server/controllers/bookmarkController.js]: You're providing one value in the array. Why do you need the [Op.or]?

File: server/controllers/bookmarkController.js [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+   * @returns {Object} Success Message
+   */
+  static async bookmarkArticle(req, res) {
+    const userId = req.user.id;
+    const { articleId } = req.params;
+    const article = await models.Article.findByPk(articleId);
+    if (!article) return utils.errorStat(res, 404, 'Article not found');
+    const bookmarked = await models.Bookmark.findOne({
+      where: {
+        [Op.or]: [{ articleId }]
```
Comment (chuxmykel) [inline server/controllers/bookmarkController.js]: Here too.

**Review by chuxmykel — APPROVED** [review]
Nice Job Martins. Please take a look at my comments and you're good to go.

File: server/controllers/bookmarkController.js (hunk 2 above)
Comment (obayomi96) [reply server/controllers/bookmarkController.js]: Thank you very much for that question @chuxmykel. It's been refactored.

File: server/docs/ah-commando-doc.yml [hunk 4]
```diff
@@ -139,7 +139,7 @@ paths:
   /users/google:
     get:
       tags:
-        - Social user
+        - Users
```
Comment (Hadeneekeh) [inline server/docs/ah-commando-doc.yml:142]: 👍🏽However, can you confirm that the same change is not needed for the facebook login?

**Review by Hadeneekeh — APPROVED** [review]
Great job @obayomi96!

Aside from @chuxmykel's comment, everything looks good to me!

File: server/docs/ah-commando-doc.yml (hunk 4 above)
Comment (obayomi96) [reply server/docs/ah-commando-doc.yml:142]: Thanks

**Review by cvjude — APPROVED** [review]
LGTM

File: server/controllers/bookmarkController.js [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    const { articleId } = req.params;
+    const article = await models.Article.findByPk(articleId);
+    if (!article) return utils.errorStat(res, 404, 'Article not found');
+    const bookmarked = await models.Bookmark.findOne({ where: { articleId } });
+    if (bookmarked) return utils.errorStat(res, 409, 'Aticle already bookmarked');
+    const addBookmark = await models.Bookmark.findOrCreate({
+      where: { articleId, userId },
+      defaults: { articleId, userId }
+    });
+    return utils.successStat(res, 201, 'Article bookmarked successfully', addBookmark);
```
Comment (topseySuave) [inline server/controllers/bookmarkController.js]: Can you change the property name for the response @obayomi96

**Review by topseySuave — APPROVED** [review]

**Review by Lundii — APPROVED** [review]
LGTM

File: server/controllers/bookmarkController.js (hunk 5 above)
Comment (obayomi96) [reply server/controllers/bookmarkController.js]: Done that now.

**Review by topseySuave — APPROVED** [review]

**Pull request merged**